packages/flutter/lib/src/rendering/editable.dart

```diff
@@ -1320,10 +1320,18 @@ class RenderEditable extends RenderBox {
       final TextPosition toPosition = to == null
         ? null
         : _textPainter.getPositionForOffset(globalToLocal(to - _paintOffset));
+
+      int baseOffset = fromPosition.offset;
+      int extentOffset = fromPosition.offset;
+      if (toPosition != null) {
+        baseOffset = math.min(fromPosition.offset, toPosition.offset);
+        extentOffset = math.max(fromPosition.offset, toPosition.offset);
+      }
```

Comment:
I'm assuming the cursor positioning does not need adjustment?

Reply:
`baseOffset` and `extendOffset` represent the cursor (or the selection) state. When `baseOffset === extentOffset`, the cursor is displayed at that position. When they are different, then they represent a range selection and Flutter will draw a selection box instead.

Does that answer your question?

Reply:
From offline discussion. Editables do remember the position of the cursor even if it's not rendered. So if you Shift + Arrow to shrink or extend the selection it will move the side corresponding to the "extent". It's OK if we don't support it now, but we should track it somewhere. So maybe file an issue, and ping @gspencergoog so it doesn't get lost.